src/harness/virtualFileSystem.ts

```diff
@@ -157,9 +159,51 @@ namespace Utils {
 
             return directory;
         }
+
+        /**
+         * Reads the directory at the given path and retrieves a list of file names and a list
+         * of directory names within it. Suitable for use with ts.matchFiles()
+         * @param path  The path to the directory to be read
+         */
+        getAccessibleFileSystemEntries(path: string) {
+            const entry = this.traversePath(path);
+            if (entry && entry.isDirectory()) {
+                const directory = <VirtualDirectory<T>>entry;
+                return {
+                    files: ts.map(directory.getFiles(), f => f.name),
+                    directories: ts.map(directory.getDirectories(), d => d.name)
+                };
+            }
+            return { files: [], directories: [] };
+        }
+
+        getAllFileEntries() {
+            const fileEntries: VirtualFile<T>[] = [];
+            getFilesRecursive(this.root, fileEntries);
+            return fileEntries;
+
+            function getFilesRecursive(dir: VirtualDirectory<T>, result: VirtualFile<T>[]) {
+                const files = dir.getFiles();
+                const dirs = dir.getDirectories();
+                for (const file of files) {
+                    result.push(file);
+                }
+                for (const subDir of dirs) {
+                    getFilesRecursive(subDir, result);
+                }
+            }
+        }
+
+        normalizePathRoot(path: string) {

```

Comment:
Whoops, that was from an old version of the code and I forgot to take it out. I'll remove it.